sdk/identity/azure-identity-v2/src/main/java/com.azure.identity.v2/CredentialAuthenticationException.java

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.v2;
+
+
+import io.clientcore.core.credentials.AccessToken;
+import io.clientcore.core.credentials.TokenCredential;
+
+/**
+ * The exception thrown when a {@link TokenCredential} did not attempt to authenticate and retrieve {@link AccessToken},
+ * as its prerequisite information or state was not available.
+ *
+ * @see com.azure.identity.v2
+ */
+public class CredentialAuthenticationException extends RuntimeException {
```

Comment:
Actually I think this is new.. Is this intended to replace `ClientAuthenticationException`?

Reply:
Yeah, the intent is to replace ClientAuthException.
We made a mistake last time, ClientAuthException extends from HttpResponseException, but we don't have the Http error info as that's msal layer, so upstream http response was null and caused NPE for users.